communications/models.py

```diff
@@ -1,3 +1,192 @@
+
+    def __str__(self):
+        return self.name
+
+    @classmethod
+    def get_default_pk(cls):
+        """Get or create a default notification type to use as a fallback.
+        
+        This is useful for database migrations and testing.
+        
+        Returns:
+            int: The primary key of the default notification type
+        """
+        obj, _ = cls.objects.get_or_create(
+            code='default',
+            defaults={
+                'name': 'Default Notification',
+                'description': 'Default notification type',
+            }
+        )
+        return obj.pk
+
+
+class Notification(models.Model):
+    """Model to store all notifications sent to users."""
+    notification_id = models.UUIDField(unique=True, editable=False, primary_key=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name='notifications'
+    )
+    notification_type = models.ForeignKey(
+        NotificationType,
+        on_delete=models.PROTECT,
+        related_name='notifications'
+    )
+    title = models.CharField(max_length=255)
+    message = models.TextField()
+    
+    triggered_by_user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name='triggered_notifications'
+    )
+    triggered_by_type = models.CharField(
+        max_length=20,
+        choices=[
+            ('investor', _('Investor')),
+            ('startup', _('Startup')),
+            ('system', _('System'))
+        ],
+        null=True,
+        blank=True
+    )
+    
+    related_startup_id = models.PositiveIntegerField(null=True, blank=True)
+    related_project_id = models.PositiveIntegerField(null=True, blank=True)
+    related_message_id = models.PositiveIntegerField(null=True, blank=True)
+    
+    priority = models.CharField(
+        max_length=10,
+        choices=[
+            ('low', _('Low')),
+            ('medium', _('Medium')),
+            ('high', _('High'))
+        ],
+        default='medium'
+    )
+    is_read = models.BooleanField(default=False)
+    expires_at = models.DateTimeField(null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ['-created_at']
+        indexes = [
+            models.Index(fields=['user', 'is_read', 'created_at'], name='idx_notif_user_read'),
+            models.Index(fields=['user', 'notification_type', 'created_at'], name='idx_notif_user_type'),
+            models.Index(fields=['triggered_by_user'], name='idx_notif_triggered_by'),
+            models.Index(fields=['related_startup_id'], name='idx_notif_startup'),
+            models.Index(fields=['related_project_id'], name='idx_notif_project'),
+            models.Index(fields=['expires_at'], name='idx_notif_expires'),
+        ]
```

Comment:
You have a lot of indexes; review your most common queries to avoid over-indexing and slowing down writes.